src/expr/impl/src/scalar/to_jsonb.rs

```diff
@@ -0,0 +1,188 @@
+    ))
+}
+
+#[function("to_jsonb(timestamptz, varchar) -> jsonb")]
+pub fn timestamptz_to_jsonb(input: Option<Timestamptz>, zone: Option<&str>) -> Result<JsonbVal> {
+    match input {
+        Some(inner) => zone
+            .ok_or_else(|| {
+                ExprError::Internal(anyhow!(
+                    "to_char(timestamptz, varchar) have a null zone_str"
+                ))
+            })
+            .and_then(|zone_str| {
+                let value = timestamptz_to_value(inner, zone_str)?;
+                Ok(value.into())
+            }),
+        None => Ok(JsonbVal::null()),
+    }
+}
+
+#[function("to_jsonb(struct, varchar) -> jsonb")]
+pub fn struct_to_jsonb(
+    input: Option<StructRef<'_>>,
+    zone: Option<&str>,
+    ctx: &Context,
+) -> Result<JsonbVal> {
+    match input {
+        Some(inner) => zone
+            .ok_or_else(|| {
+                ExprError::Internal(anyhow!("to_char(struct, varchar) have a null zone_str"))
+            })
+            .and_then(|zone_str| {
+                let value = struct_to_value(inner, zone_str, &ctx.arg_types[0])?;
+                Ok(value.into())
+            }),
+        None => Ok(JsonbVal::null()),
+    }
+}
+
+#[function("to_jsonb(anyarray, varchar) -> jsonb")]
+pub fn list_to_jsonb(
+    input: Option<ListRef<'_>>,
+    zone: Option<&str>,
+    ctx: &Context,
+) -> Result<JsonbVal> {
+    match input {
+        Some(inner) => zone
+            .ok_or_else(|| {
+                ExprError::Internal(anyhow!("to_char(anyarray, varchar) have a null zone_str"))
+            })
+            .and_then(|zone_str| {
+                let value = anyarrary_to_value(inner, zone_str, &ctx.arg_types[0])?;
+                Ok(value.into())
+            }),
+        None => Ok(JsonbVal::null()),
+    }
+}
+
+fn timestamptz_to_value(input: Timestamptz, zone: &str) -> Result<Value> {
+    let time_zone = Timestamptz::lookup_time_zone(zone).map_err(time_zone_err)?;
+    let instant_local = input.to_datetime_in_zone(time_zone);
+    Ok(instant_local.to_rfc3339().as_bytes().into())
+}
+
+fn anyarrary_to_value(input: ListRef<'_>, zone: &str, data_type: &DataType) -> Result<Value> {
+    let data_type_in_list = data_type.as_list();
+    let values: Vec<Value> = input
+        .iter()
+        .map(|x| scalar_to_value(x, zone, data_type_in_list))
+        .try_collect()?;
+    Ok(Value::array(values.iter().map(|v| v.as_ref())))
+}
+
+fn struct_to_value(input: StructRef<'_>, zone: &str, data_type: &DataType) -> Result<Value> {
+    let mut builder = jsonbb::Builder::default();
+    builder.begin_object();
+
+    let names: Vec<&str> = data_type.as_struct().names().collect();
+    input
+        .iter_fields_ref()
+        .zip_eq_fast(data_type.as_struct().types())
+        .enumerate()
+        .map(|(idx, (scalar, data_type))| {
+            let key = if names.is_empty() {
+                format!("f{}", idx + 1)
+            } else {
+                names[idx].to_string()
+            };
```

Comment:
use `builder.display(impl Display)` API to avoid generating String.

```suggestion
            let key = if names.is_empty() {
                builder.display(format_args!("f{}", idx + 1));
            } else {
                builder.add_string(names[idx]);
            };
```

PS. maybe I should rename `display` to `add_string_display` for consistency. 🤔

Reply:
done